xpm/xpm.go

```diff
@@ -50,20 +51,46 @@ func parseXPM(data io.Reader) (image.Image, error) {
 }
 
 func parseColor(data string, charSize int) (id string, c color.Color, err error) {
-	if len(data) == 0 {
+	// XXX: Error handling is lacking in this function.
```

Comment:
I don't fully understand this comment - can't you return an error using the parameter there already?

Reply:
I kept the error handling as it was in this pull request because I was not sure whether these default values were intentional, and did not want to touch it.

But I do have a series of follow-up patches which add a fuzzer to catch the remaining crashes, and in the course of which this error handling is also getting fixed.  So I'd address it in that follow-up, if that's ok with you.

Reply:
I see. Perhaps that comment should be more specific about not wanting to return an error in the specific case it is attached to? As you can see the err return is set further into the method in certain situations.

Reply:
I made the comment a bit longer to clarify, even though I believe it'll soon be fixed and removed, assuming that you are OK with the follow-up patch set.  I pushed that one to https://github.com/fyne-io/image/compare/main...gnoack:fyne-image:picons2 if you want to have an early look. (TL;DR: I threw a fuzzer at the XPM parser to catch the remaining crashes).

Reply:
> Please take another look. I made the comment a bit longer to clarify, even though I believe it'll soon be fixed and removed

Thanks. That may seem like a peculiar request but we don't know how long it will be between PRs.
If it was a note to self you could have not pushed the comment at all ;).